src/BTCPayServer.Lightning.Charge/ChargeClient.cs

```diff
@@ -49,24 +39,54 @@ public ChargeClient(Uri uri, Network network, HttpClient httpClient)
                 throw new ArgumentException(paramName: nameof(uri), message: "User information not present in uri");
             ChargeAuthentication = new ChargeAuthentication.UserPasswordAuthentication(new NetworkCredential(userInfo[0], userInfo[1]));
         }
-        public ChargeClient(Uri uri, string cookieFilePath, Network network): this(uri, cookieFilePath, network, null)
-        {
 
-        }
-        public ChargeClient(Uri uri, string cookieFilePath, Network network, HttpClient httpClient)
+        public ChargeClient(Uri uri, string cookieFilePath, Network network, HttpClient httpClient = null, bool allowInsecure = false)
         {
             if (uri == null)
                 throw new ArgumentNullException(nameof(uri));
             if (network == null)
                 throw new ArgumentNullException(nameof(network));
             if (cookieFilePath == null)
                 throw new ArgumentNullException(nameof(cookieFilePath));
-            httpClient = httpClient ?? _SharedClient;
+            httpClient = CreateHttpClient(uri, allowInsecure, httpClient);
```

Comment:
should be `httpClient ?? _SharedClient`

Reply:
Yes, it reuses the same httpclient when possible. See https://github.com/btcpayserver/BTCPayServer.Lightning/pull/35/files#diff-187e3bd22467c64f3d1c1103ac220062R60

Reply:
it does not, you don't use `_SharedClient` anywhere.

Reply:
fixed